daemon/src/position_metrics.rs

```diff
@@ -211,52 +210,44 @@ impl Cfd {
                 ..self
             },
             CetConfirmed => Self {
-                is_open: false,
-                is_closed: true,
+                state: AggregatedState::Closed,
                 ..self
             },
             RefundConfirmed => Self {
-                is_open: false,
-                is_refunded: true,
+                state: AggregatedState::Closed,
                 ..self
             },
             RevokeConfirmed => Self {
                 // the other party was punished, we are done here!
-                is_open: false,
-                is_closed: true,
+                state: AggregatedState::Closed,
                 ..self
             },
             CollaborativeSettlementConfirmed => Self {
-                is_open: false,
-                is_closed: true,
+                state: AggregatedState::Closed,
                 ..self
             },
             LockConfirmedAfterFinality => Self {
                 // This event is only appended if lock confirmation happens after we spent from lock
                 // on chain. This is for special case where collaborative settlement is triggered
                 // before lock is confirmed. In such a case the CFD is closed
-                is_open: false,
-                is_closed: true,
+                state: AggregatedState::Closed,
```

Comment:
🤔 It seems weird to overwrite anything here. I would rely on whatever happened before and not do anything here.

Reply:
True, we can just remain in whatever state we have been.